scripts/utils/setup-roles.ts

```diff
@@ -1,8 +1,11 @@
 const fs = require("fs");
-import hre, { upgrades, ethers } from "hardhat";
+import hre, { ethers } from "hardhat";
 import { Contract } from "ethers";
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
 
+// This is imported to extend into this file: import "@nomiclabs/hardhat-ethers";
```

Comment:
Is there a reason we don't just use `import @nomiclabs/hardhat-ethers` directly?

Reply:
If we import it directly, hardhat context node modules complain:

` throw new HardhatError(ERRORS.GENERAL.CONTEXT_NOT_CREATED);`

Reply:
What command are you running exactly?

Reply:
HARDHAT_NETWORK=rinkeby npm run setup-roles rinkeby 1655327670025000

Reply:
Ah - use `npx hardhat run` instead of `ts-node` in the `package.json`. Then you don't need this import. `npx hardhat run` would be standard for anything that requires us to import/use our contracts. I also think it's more standard to use the `--network` tag as opposed to `HARDHAT_NETWORK`.

So again, try doing my suggested change, change the package.json to use `npx hardhat run` instead of `ts-node`. and then:

`npm run setup-roles 1655327670025000 --network rinkeby`

(note I'm also removing the double use of `rinkeby` - looks redundant. You could grab the network to look for the correct filename in the script using `hre.network.name`.

Reply:
I tried the way you are suggesting, but the error in the screenshot is generated. 
It seems that hardhat's run task does not support forwarding any parameters that it doesn't recognize as arguments to a script.

![Screen Shot 2022-06-16 at 11 11 36 AM](https://user-images.githubusercontent.com/42501101/174102088-9959b935-bd19-45a7-9ee0-deed6d9b1006.png)

The solution would be to create a "task" for this in `hardhat.config.ts ` per this
[example](https://github.com/NomicFoundation/hardhat/issues/2292).

Isn't that a lot more code to just avoid one import?!

Happy to go either way.  Pls let me know your thoughts!

Reply:
You can leave it as is for now, but I will clean this up in my PR.....it's a code smell. We should do the extra work to not just make it work, but make it work in an elegant way